active/TreeView/TreeViewSelectionAPI.md

```diff
@@ -0,0 +1,76 @@
+# Background
+
+`SelectedNodes` is the only API we currently have on TreeView to update selections, and it only works in multi-select mode, which means there's no easy way to programmatically change selections in single selection mode. Also, with the new data binding capability, people would prefer to work on their own data source instead of `TreeViewNode`. With current APIs, they will have to do `ContainerFromItem` and then `NodeFromContainer` to get the associated TreeViewNode in order to update SelectedNodes.

```

Comment:
I'm not following the last part; what is the new alternative to ContainerFromItem and NodeFromContainer?

Reply:
They can just use the new API, TreeView.SelectedItem(s).

Reply:
Our old API sets operate on TreeViewNode. If a developer uses data binding, they'll have to convert back and forth between the ItemsSource type and TreeViewNode to update selections. 

With the new API they can update selection by directly passing in the data source type objects.

Reply:
So if you're using ItemsSource, and you want to select an item, you have to do 
```
treeView.SelectedNodes.Clear()
treeView.SelectedNode.Add( NodeFromContainer(ContainerFromItem(item)) );
```
?

Reply:
The proposed API shows it would be this to select an item:
```
treeView.SelectedItem = item;
```

Reply:
Right. I just couldn't figure out why the text was talking about those other APIs. But I see now that what it's saying is

Old code:
```
treeView.SelectedNodes.Clear()
treeView.SelectedNode.Add( NodeFromContainer(ContainerFromItem(item)) );
```
New code:
```
treeView.SelectedItem = item;
```
I'm working on some updates in a branch, I'll include this explanation.